src/main/java/Duke.java

```diff
@@ -1,10 +1,190 @@
+
+    private static int dividerPosition;
+    private static int charAfterDividerPosition;
+
+    private static final String COMMAND_LIST = "list";
+    private static final String COMMAND_DONE = "done";
+    private static final String COMMAND_BYE = "bye";
+    private static final String COMMAND_TODO = "todo";
+    private static final String COMMAND_DEADLINE = "deadline";
+    private static final String COMMAND_EVENT = "event";
+
+
+    private static void printLogo() {
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
     }
+
+    private static void printGreeting(String s, String s2) {
+        System.out.println(HORIZONTAL_LINE);
+        System.out.println(s);
+        System.out.println(s2);
+    }
+
+    private static String getUserInput() {
+        System.out.println(HORIZONTAL_LINE);
+        System.out.println("Enter command: ");
+        Scanner in = new Scanner(System.in);
+        return in.nextLine();
+    }
+
+    private static void initTaskList() {
+        tasks = new Task[MAX_TASK_LIMIT];
+    }
+
+    private static void processUserInput() {
+        String[] splitInput = userInput.split(" ");
+        String inputCommand = splitInput[0];
+        switch (inputCommand) {
+        case COMMAND_LIST:
+            performListTask();
+            break;
+        case COMMAND_DONE:
+            handleIntConversion(splitInput[1]);
+            performMarkTaskDone(inputNum);
+            break;
+        case COMMAND_TODO:
+            splitTodo(userInput);
+            performAddTodo(todoTask);
+            break;
+        case COMMAND_DEADLINE:
+            splitDeadline(userInput);
+            performAddDeadline(deadlineDescription, by);
+            break;
+        case COMMAND_EVENT:
+            splitEvent(userInput);
+            performAddEvent(eventDescription, at);
+            break;
+        case COMMAND_BYE:
+            exitProgram();
+            break;
+        default:
+            addTask();
+        }
+    }
+
+    private static void addTask(){
+        Task t = new Task(userInput);
+        tasks[Task.taskCount - 1] = t;
+        System.out.println(HORIZONTAL_LINE);
+        System.out.println("added: " + userInput.trim());
+        System.out.println("Now you have " + Task.taskCount + " tasks in the list");
+    }
+
+    private static void performListTask() {
+        System.out.println(HORIZONTAL_LINE);
+        System.out.println("Here are the tasks in your list:");
+
+        for (int i = 0; i < Task.taskCount; i++) {
+            System.out.println(INDENT + tasks[i].taskNum + "." + tasks[i]);
+        }
+    }
+
+    private static void handleIntConversion(String number){
+        number = number.trim();
+        inputNum = Integer.parseInt(number);
```

Comment:
Recommend to place this line within try catch just in case the string provided is not an integer.